### ISSUE TITLE
False positive: Unnecessary type casting

### ISSUE THREAD

**Issue by KriqN** [kalessil/phpinspectionsea#1669]
PhpStorm 2020.3
Plugin 4.0.6
PHP 7.3

```
$price = '250.72';
$priceInt = (int) ($price * 100);
```
So `$price * 100` returns float, but inspection is reporting "This type casting is not necessary, as the argument is of needed type".

![EA float int](https://user-images.githubusercontent.com/63552810/101750557-d7edd600-3ad7-11eb-8769-63348697cf47.png)

_Originally posted by @KriqN in https://github.com/kalessil/phpinspectionsea/issues/490#issuecomment-742385915_

**Comment by ea-inspections-team** [comment]
Thanks for reporting @KriqN ! We'll look into how to fix this.

**Comment by ea-inspections-team** [comment]
Thanks for reporting @KriqN! Fixed!
